File: classes/game/item/JellyAnimatorWaiter.cpp
#include "cocos2d.h"
#include "JellyAnimatorWaiter.h"
#include "JellyItem.h"


void JellyAnimatorWaiter::InformAnimationEnded() {
	//this->SpreadAnimator.SetInteger("DIRECTION", 5);
	if (this->MasterJelly != nullptr) {
		this->MasterJelly->AnimationEnded();
	}
}

void JellyAnimatorWaiter::DisplayAtFront() {
}

void JellyAnimatorWaiter::HideIt() {
}

File: doc/porting/JellyAnimatorWaiter.cs
using System;
using UnityEngine;

namespace Assets.Scripts.GamePlayScene.Mechanics.Items
{
	public class JellyAnimatorWaiter : MonoBehaviour
	{
		[HideInInspector]
		public JellyItem MasterJelly;

		public Animator SpreadAnimator;

		public Renderer SpreadingSprite;

		public int DisplaySortNo;

		public void InformAnimationEnded()
		{
			this.SpreadAnimator.SetInteger("DIRECTION", 5);
			if (this.MasterJelly != null)
			{
				this.MasterJelly.AnimationEnded();
			}
		}

		public void DisplayAtFront()
		{
			this.SpreadingSprite.set_sortingLayerName("Mechanics");
			this.SpreadingSprite.set_sortingOrder(this.DisplaySortNo);
		}

		public void HideIt()
		{
			this.SpreadingSprite.set_sortingLayerName("Default");
			this.SpreadingSprite.set_sortingOrder(-10);
		}
	}
}
